src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,163 @@
+                this::setTargetAngleFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private Command getSetTargetElevatorPositionCommand(double targetElevatorPosition, double speedPercentage) {
+        return new FunctionalCommand(
+                () -> generateElevatorMotorProfile(targetElevatorPosition, speedPercentage),
+                this::setTargetElevatorPositionFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private void generateAngleMotorProfile(Rotation2d targetAngle, double speedPercentage) {
+        angleMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ANGLE_CONSTRAINTS, speedPercentage),
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAnglePositionDegrees().getDegrees(), getAngleVelocityDegreesPerSecond())
+        );
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void generateElevatorMotorProfile(double targetElevatorPosition, double speedPercentage) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ELEVATOR_CONSTRAINTS, speedPercentage),
+                new TrapezoidProfile.State(targetElevatorPosition, 0),
+                new TrapezoidProfile.State(armInputs.elevatorPositionRevolutions, armInputs.elevatorVelocityRevolutionsPerSecond)
+        );
+        lastElevatorMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            armIO.stopAngleMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleProfileTime());
+        armIO.setTargetAngle(targetState);
+    }
+
+    private void setTargetElevatorPositionFromProfile() {
+        if (elevatorMotorProfile == null) {
+            armIO.stopElevatorMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorProfileTime());
+        armIO.setTargetElevatorPosition(targetState);
+    }
+
+    private double getAngleProfileTime() {
+        return Timer.getFPGATimestamp() - lastAngleMotorProfileGenerationTime;
+    }
+
+    private double getElevatorProfileTime() {
+        return Timer.getFPGATimestamp() - lastElevatorMotorProfileGenerationTime;
+    }
+
+    private Rotation2d getAnglePositionDegrees() {
+        return Rotation2d.fromRotations(armInputs.angleEncoderPositionSignal);
+    }
+
+    private double getAngleVelocityDegreesPerSecond() {
+        return Conversions.perHundredMsToPerSecond(Conversions.revolutionsToDegrees(armInputs.angleEncoderVelocitySignal));
+    }
```

Comment:
have the conversion in the io class. The inputs class should have the variables with the units you want already